lib/srv/server/ec2_watcher.go

```diff
@@ -104,8 +132,54 @@ type ec2InstanceFetcher struct {
 	Region       string
 	DocumentName string
 	Parameters   map[string]string
+
+	// cachedInstances keeps all of the ec2 instances that were matched
+	// in the last run of GetInstances for use as a cache with
+	// GetMatchingInstances
+	cachedInstances *instancesCache
+}
+
+type instancesCache struct {
+	sync.RWMutex
```

Comment:
```suggestion
	sync.Mutex
```

This is not getting enough contention to justify a `RWMutex`, a regular one will perform better.